codegen/post_gen_hooks.go

```diff
@@ -89,10 +89,10 @@ func ClientMockGenHook(h *PackageHelper, t *Template) (PostGenHook, error) {
 		for _, instance := range clientInstances {
 			key := instance.ClassType + instance.InstanceName
 			var mc moduleConfig
-			if err := json.Unmarshal(instance.JSONFileRaw, &mc); err != nil {
+			if err := yaml.Unmarshal(instance.YAMLFileRaw, &mc); err != nil {
 				return errors.Wrapf(
 					err,
-					"error parsing client-config.json for client %q",
+					"error parsing client-config.yaml for client %q",
```

Comment:
nit: Since it could be either `client-config.json` or `client-config.yaml` you should use the import path here rather than hard coding it.